samcli/commands/pipeline/bootstrap/guided_context.py

```diff
@@ -0,0 +1,94 @@
+        ecr_repo_arn: Optional[str] = None,
+        pipeline_ip_range: Optional[str] = None,
+    ) -> None:
+        self.stage_name = stage_name
+        self.pipeline_user_arn = pipeline_user_arn
+        self.pipeline_execution_role_arn = pipeline_execution_role_arn
+        self.cloudformation_execution_role_arn = cloudformation_execution_role_arn
+        self.artifacts_bucket_arn = artifacts_bucket_arn
+        self.create_ecr_repo = create_ecr_repo
+        self.ecr_repo_arn = ecr_repo_arn
+        self.pipeline_ip_range = pipeline_ip_range
+
+    def run(self) -> None:
+        """
+        Runs an interactive questionnaire to prompt the user for the ARNs of the AWS resources(infrastructure) required
+        for the pipeline to work. Users can provide all, none or some resources' ARNs and leave the remaining empty
+        and it will be created by the bootstrap command
+        """
+        if not self.stage_name:
+            self.stage_name = click.prompt("Stage Name", type=click.STRING)
+
+        if not self.pipeline_user_arn:
+            click.echo(
+                "\nThere must be exactly one pipeline user across all of the pipeline stages. "
+                "If you have ran this command before to bootstrap a previous pipeline stage, please "
+                "provide the ARN of the created pipeline user, otherwise, we will create a new user for you, "
+                "please make sure to configure user's AccessKeyId and SecretAccessKey for the CI/CD provider."
+            )
+            self.pipeline_user_arn = click.prompt(
+                "Pipeline user [leave blank to create one]", default="", type=click.STRING
+            )
+
+        if not self.pipeline_execution_role_arn:
+            self.pipeline_execution_role_arn = click.prompt(
+                "\nPipeline execution role(an IAM Role to be assumed by the pipeline-user to operate on this stage.) "
+                "[leave blank to create one]",
+                default="",
+                type=click.STRING,
+            )
+
+        if not self.cloudformation_execution_role_arn:
+            self.cloudformation_execution_role_arn = click.prompt(
+                "\nCloudFormation execution role(an IAM Role to be assumed by the CloudFormation service to deploy "
+                "the application's stack) [leave blank to create one]",
+                default="",
+                type=click.STRING,
+            )
+
+        if not self.artifacts_bucket_arn:
+            self.artifacts_bucket_arn = click.prompt(
+                "\nArtifacts bucket(S3 bucket to hold the sam build artifacts) " "[leave blank to create one]",
+                default="",
+                type=click.STRING,
+            )
+        if not self.ecr_repo_arn:
+            click.echo(
+                "\nIf your SAM template will include lambda functions of Image package-type, "
+                "then an ECR repo is required, should we create one?"
```

Comment:
```suggestion
                "\nIf your SAM template will include Lambda functions of Image package type, "
                "then an ECR repository is required. Should we create one?"
```

Reply:
I used the future tense to let the users think about the future; Instead of thinking "no my templates doesn't contain this" they will think "will my template contain this".

Reply:
but it is not how "if" is used in English as far as I know. You can check the article I posted above, they cover all scenarios and none of them use future tense. (specifically about the "likely")

Reply:
more reading: https://www.grammarly.com/blog/will-would-in-if-clause/